aea/configurations/base.py

```diff
@@ -1477,17 +726,32 @@ def json(self) -> Dict:
         return {"class_name": self.class_name, "args": self.args}
 
     @classmethod
-    def from_json(cls, obj: Dict):
+    def from_json(cls, obj: Dict) -> "SkillComponentConfiguration":
+        """Initialize from a JSON object."""
+        return cls._create_or_update_from_json(obj, instance=None)
+
+    @classmethod
+    def _create_or_update_from_json(
+        cls, obj: Dict, instance: Optional["SkillComponentConfiguration"] = None
+    ) -> "SkillComponentConfiguration":
         """Initialize from a JSON object."""
         class_name = cast(str, obj.get("class_name"))
-        return SkillComponentConfiguration(class_name=class_name, **obj.get("args", {}))
+        params = dict(class_name=class_name, **obj.get("args", {}))
+
+        if instance is None:
+            instance = cls(**params)  # type: ignore
+        else:
+            instance.__init__(**params)  # type: ignore
+
+        return instance
```

Comment:
repeat code on several pages, suggests moving to base class

Reply:
done